app/src/integrationTests/ReportIntegrationTests.tsx

```diff
@@ -0,0 +1,189 @@
+    }
+
+    addTestsToMocha() {
+
+        const sandbox = new Sandbox();
+        const fetcher = new ReportingFetcher();
+
+        afterEach(() => sandbox.restore());
+
+        it("fetches reports", (done: DoneCallback) => {
+            const promise = reportStore.fetchReports();
+            const expected: string[] = ["minimal", "multi-artefact", "multifile-artefact", "other", "use_resource"];
+
+            checkPromise(done, promise, (reports) => expectSameElements<string>(reports, expected));
+        });
+
+        it("fetches report versions", (done: DoneCallback) => {
+            reportActions.setCurrentReport("minimal");
+            const promise = reportStore.fetchVersions();
+
+            checkPromise(done, promise, (versions) => expect(versions.length).to.be.greaterThan(0));
+        });
+
+        it("fetches report version details", (done: DoneCallback) => {
+            reportActions.setCurrentReport("minimal");
+            const promise = reportStore.fetchVersions().then((versions) => {
+                reportActions.setCurrentVersion(versions[0]);
+                return reportStore.fetchVersionDetails()
+            });
+
+            checkPromise(done, promise, (versionDetails) => expectIsEqual(versionDetails.name, "minimal"));
+        });
+
+        it("fetches report one time token", (done: DoneCallback) => {
+
+            reportActions.setCurrentReport("minimal");
+            let version = "";
+
+            const promise = reportStore.fetchVersions()
+                .then((versions) => {
+                    version = versions[0];
+
+                    reportActions.setCurrentVersion(version);
+                    return reportStore.fetchVersionDetails()
+                        .then((details: Version) =>
+                            oneTimeTokenStore.fetchToken(`/reports/minimal/${version}/artefacts/`))
+                });
+
+            checkPromise(done, promise, token => {
+                const decoded = jwt_decode(token);
+                expect(decoded.url).to.equal(`/v1/reports/minimal/${version}/artefacts/`);
+            });
+        });
+
+        it("downloads artefact", (done: DoneCallback) => {
+
+            const reportName = "multi-artefact";
+            const promise = getVersion(reportName)
+                .then((version: string) => {
+
+                    const rendered = shallow(<ArtefactItem report={reportName} version={version}
+                                                           filenames={["all.csv", "all.png"]}
+                                                           description="all things"/>);
+
+                    const link = rendered.find(FileDownloadLink).first();
+
+                    const url = link.prop("href");
+                    return fetcher.fetchFromReportingApi(url)
+
+                });
+
+            checkPromise(done, promise, (response) => {
+                expect(response.status).to.equal(200)
+            });
+        });
```

Comment:
In each of these file download tests there are some repeated lines that could be factored out:

```
const link = rendered.find(FileDownloadLink).first();
const url = link.prop("href");
return fetcher.fetchFromReportingApi(url)
```

and 

```
checkPromise(done, promise, (response) => {
    expect(response.status).to.equal(200)
});
```